editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/pages/FuseIntegrationProjectWizardTemplatePage.java

```diff
@@ -195,6 +197,15 @@ private FilteredTree createFilteredTree(Composite parent) {
 		list_templates.setLayoutData(GridDataFactory.swtDefaults().align(SWT.FILL, SWT.FILL).create());
 		list_templates.getViewer().setContentProvider(new TemplateContentProvider());
 		list_templates.getViewer().setLabelProvider(new TemplateLabelProvider());
+		list_templates.getViewer().addFilter(new ViewerFilter() {	
```

Comment:
I would extract ViewerFilter in a dedicated File (which will bring the isEmptyCategory methods too) and then be able to write easily a little unit test on the filter